### PR TITLE
fix(tagsInput): fix add-on-paste issue in IE

### LINKED ISSUE #325: Pasting in IE fails to work with add-on-paste
0. Install a Windows OS of some kind that uses IE 10 or 11.
1. Use ng-tags-input with the option `add-on-paste="true"`.
2. In IE, paste the following into the input: `foo,bar,bazz`
3. Expected behavior is for `foo`, `bar`, and `bazz` to exist as individual tags. Outcome is `foo` is in the input, and there are no tags.

Additionally, this error appears in the console:
`TypeError: Unable to get property 'getData' of undefined or null reference`

### PR DESCRIPTION
**Pull request by LoganBarnett**
Fix an error when pasting in IE. IE uses window.clipboardData
instead of putting clipboardData on the event itself.

Fixes #325.

### CODE REVIEW

File: .jshintrc [hunk 1]
```diff
@@ -40,6 +40,7 @@
         "customMatchers": true,
         "KEYS": true,
         "MAX_SAFE_INTEGER": true,
-        "SUPPORTED_INPUT_TYPES": true
+        "SUPPORTED_INPUT_TYPES": true,
+        "window": true
```
Comment (mbenford) [inline .jshintrc]: You should use Angular's `$window` service, instead of the global variable `window`.

**Comment by mbenford** [comment]
Could you adjust your PR according to the [CONTRIBUTING](https://github.com/mbenford/ngTagsInput/blob/master/CONTRIBUTING.md) file?

**Comment by LoganBarnett** [comment]
Not sure what I missed, but I'll take a look!

I'll update for $window too. Thanks!

**Comment by mbenford** [comment]
Any update on this?

**Comment by LoganBarnett** [comment]
Thanks for the prod. I got the PR using $window now. If there's something I'm missing from the [CONTRIBUTING](https://github.com/mbenford/ngTagsInput/blob/master/CONTRIBUTING.md) file I haven't been able to tell what it is. Could I get some help in that regard? :)

**Comment by mbenford** [comment]
You need to write some test(s) to cover your changes. If you take a look at Coveralls messages above, you'll see that coverage has decreased, which means tests are missing.

**Comment by mbenford** [comment]
The [coverage report](https://coveralls.io/builds/2110474) can help you find out what needs to be covered.

**Comment by LoganBarnett** [comment]
I have absolutely no problem getting this covered in the tests. As you mentioned in #325 the whole thing gets mocked out so it might not be a great test. I believe you're correct that Angular should just abstract this completely rather than partially, and there should be a PR sent their way. I don't mind heading that up.

In the mean time, a test for the sake of coverage reports? :)

**Comment by LoganBarnett** [comment]
Ok should be ready to roll with the IE fix. Apologies for the mail spam - just had some really bad luck with git today. All worked out now though.

**Comment by mbenford** [comment]
Thanks! :+1: 

What I said above wasn't about coverage numbers. You have added new code that doesn't get hit by any tests, which means I can change that code or even delete it altogether and nothing is going to break, which in turn makes my build script less reliable. I simply mentioned that coverage had decreased because that made it clear something was missing.

Having all features and (especially) bugfixes covered by tests means I can confidently release a new version as soon as grunt says `Done, without errors.`. :)

**Comment by LoganBarnett** [comment]
While I was trying to have a little bit of fun re: coverage numbers, what you've said has given me a lot to think about. I've struggled with the value of mocks (or perhaps degrees of mocking) for quite a while, and I think this was the missing piece for me, as silly as that may sound.

Thanks!

**Pull request closed without merging**